### PR TITLE
{*}[intel/2016.00] HPL 2.1 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
requires https://github.com/hpcugent/easybuild-easyblocks/issues/751 and https://github.com/hpcugent/easybuild-easyblocks/pull/773

wrapper PR for https://github.com/hpcugent/easybuild-easyconfigs/pull/1973 which includes https://github.com/DirkdeDraak/easybuild-easyconfigs/pull/1, and list the components for icc/ifort to avoid a big fat installation that also includes mkl/tbb/ipp/etc

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 10 out of 11 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/3ca11cfd71dd7d25dceb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/0d14aabf37bccca520cd for a full test report.

**Comment by wpoely86** [comment]
I think you mean `2016a`?

File: easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+checksums = ['f57a892fb494db3c80f20a88aa3e901f']
+
+gcc = 'GCC'
+gccver = '4.9.3'
+versionsuffix = '-%s-%s' % (gcc, gccver)
+
+dependencies = [(gcc, gccver)]
+
+components = ['intel-comp.*', 'intel-ccomp.*', 'intel-icc-.*', 'intel-openmp.*', 'intel-ipsc?.*']
```
Comment (wpoely86) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb]: the `.*` are not needed?

File: easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+homepage = 'http://software.intel.com/en-us/intel-compilers/'
+description = "C and C++ compiler from Intel"
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+sources = ['parallel_studio_xe_%(version_major)s_composer_edition_for_cpp.tgz']
+
+checksums = ['f57a892fb494db3c80f20a88aa3e901f']
+
+gcc = 'GCC'
```
Comment (wpoely86) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb]: I suppose this should be `GCCCore` when merged?

File: easybuild/easyconfigs/i/icc/icc-2016.0.109.eb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
+
```
Comment (wpoely86) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109.eb]: Shouldn't we add a comment that we implicietly use the system GCC here? To clearly state the difference with the previous easyconfig for new people? I would take this one if I didn't know better :wink:

File: easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb]: I also would add a comment where the full list can be found.

Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb]: yes they are, we're using `match`, not `search`

ok on the comment

File: easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb]: yeah, probably, I need to test this stuff on top of the GCCcore changes

File: easybuild/easyconfigs/i/icc/icc-2016.0.109.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109.eb]: ok, will do

File: easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb]: No, they are not needed. We only match, we don't use the result. The last regex should be `intel-ipsc?_`

Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb]: without `.*`, `match` will not... match

Comment (wpoely86) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3.eb]: Why wouldn't it? 
```python
import re
>>> print re.match("intel-comp", "intel-comp-bla")
<_sre.SRE_Match object at 0x7f7d5e42d5e0>
>>> print re.match("intel-comp.*", "intel-comp-bla")
<_sre.SRE_Match object at 0x7f7d5e42d648>
```
You start to own me a beer.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/a9f89973f642abcdff45 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/8b06d49aeadb7e39c215 for a full test report.

File: easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3-2.25.eb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+homepage = 'http://software.intel.com/en-us/intel-compilers/'
+description = "C and C++ compiler from Intel"
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+sources = ['parallel_studio_xe_%(version_major)s_composer_edition_for_cpp.tgz']
+
+checksums = ['f57a892fb494db3c80f20a88aa3e901f']
+
+gcc = 'GCC'
```
Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2016.0.109-GCC-4.9.3-2.25.eb]: as pointed out by @ocaisa in #2214, this should be `GCCcore`, and `binutils` should be included as well as a dep

File: easybuild/easyconfigs/i/iccifort/iccifort-2016.0.109-GCC-4.9.3-2.25.eb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+name = 'iccifort'
+version = '2016.0.109'
+
+homepage = 'http://software.intel.com/en-us/intel-cluster-toolkit-compiler/'
+description = """Intel Cluster Toolkit Compiler Edition provides Intel C,C++ and fortran compilers, Intel MPI and Intel MKL"""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+gcc = 'GCC'
```
Comment (boegel) [inline easybuild/easyconfigs/i/iccifort/iccifort-2016.0.109-GCC-4.9.3-2.25.eb]: as pointed out by @ocaisa in #2214, this should be `GCCcore`, and `binutils` should be included as well as a dep

File: easybuild/easyconfigs/i/ifort/ifort-2016.0.109-GCC-4.9.3-2.25.eb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+homepage = 'http://software.intel.com/en-us/intel-compilers/'
+description = "C and C++ compiler from Intel"
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+sources = ['parallel_studio_xe_%(version_major)s_composer_edition_for_fortran.tgz']
+
+checksums = ['bce7f6a71f7e44f67956197501d00b7c']
+
+gcc = 'GCC'
```
Comment (boegel) [inline easybuild/easyconfigs/i/ifort/ifort-2016.0.109-GCC-4.9.3-2.25.eb]: as pointed out by @ocaisa in #2214, this should be `GCCcore`, and `binutils` should be included as well as a dep

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (11 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/808207b15196b8f9a9b8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/81da6347c3a3af031684 for a full test report.

**Comment by boegel** [comment]
Tested thoroughly on top of https://github.com/hpcugent/easybuild-easyblocks/pull/756, good to go, thanks a lot @DirkdeDraak for your work on this!

**Comment by boegel** [comment]
And to @wpoely86 for the help in getting the `IntelBase` and `EB_icc` easyblocks up to speed. Going in!

**Pull request merged**